cmake/dev.cmake

```diff
@@ -38,6 +38,78 @@ ELSE()
         LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wextra")
         LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-pedantic")
         LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Werror")
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wdouble-promotion")
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wmissing-include-dirs")
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wunknown-pragmas")
+        # Higher levels (max is 5) produce some strange warnings
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wstrict-overflow=2")
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wtrampolines")
+        LIST(APPEND ALPAKA_DEV_COMPILE_OPTIONS "-Wfloat-equal")
```

Comment:
This flag should not enabled by default in alpaka. alpaka uses no floats where is flag is useful.
This means if the user need to fulfill this strict rule the flag can be enabled by the user.